controllers/shared.go

```diff
@@ -22,7 +23,21 @@ func nodeUpdateNeedsReconcile(ev event.UpdateEvent) bool {
 	if newNode, ok = ev.ObjectNew.(*v1.Node); !ok {
 		return false
 	}
-	return conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+
+	return labelsUpdateNeedsReconcile(oldNode.Labels, newNode.Labels) ||
+		conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+}
+
+func labelsUpdateNeedsReconcile(oldLabels, newLabels map[string]string) bool {
+	// Check if the ExcludeFromRemediation label was removed
```

Comment:
I'm wondering if it would make sense to reconcile a node with added label 🤔 It would at least give better visibility to what NHC is doing or not doing in the logs... WDYT?

Reply:
It makes sense to me (I just need to bring a logger to this function)

we could add a log when the label change is detected like:
- label added: "Label XYZ was added on Node ABC. Future remediations on this Node will be skipped
  until the label is present"
- label removed: "Label XYZ was removed on Node ABC."

I'm just not sure about the "Future remediation" part, but I don't want to give the idea that the
Node needs a remediation right now, and we are talking about "potential" future remediations.

I'd like to modify also the log

> "Skipped remediation because node ABC is marked to exclude remediations"

into something like

> "Remediation skipped on node ABC: node has remediation.medik8s.io/exclude-from-remediation label"

(or a more generic Exclude from remediation label, so that we don't need to maintain this log if we
ever need to change the label's name in the future)

Reply:
I mean returning true here in case the label was set. Reconcile we do the logging.
And +1 for improving the log message over there!

Reply:
>I mean returning true here in case the label was set.

Just to make sure I get it right.
- you only mean commonLabels.ExcludeFromRemediation in that context and not other labels 
- We will trigger the reconcile for "Excluded" nodes in order to have clearer logs/events and later on the reconcile method itself will identify the situation and stop.

Reply:
correct